product-service/handlers/addProduct.ts

```diff
@@ -0,0 +1,53 @@
+import { APIGatewayProxyHandler } from 'aws-lambda';
+import 'source-map-support/register';
+import { addProductToDb } from '../src/dataProvider';
+import { parseBodyString, validateProductData } from '../helpers';
+
+export const addProduct: APIGatewayProxyHandler = async (event, _context) => {
+  const headers = {
+    "Access-Control-Allow-Headers" : "Content-Type",
+    "Access-Control-Allow-Origin": "*",
+    "Access-Control-Allow-Methods": "OPTIONS,POST,GET"
+  };
+
+  try {
+    const data = parseBodyString(event.body);
```

Comment:
I think that `parseBodyString` didn't work as you expected. I think that you should use `JSON.parse` for this

Reply:
Format of event.body
`"count=3&title=Samsung%20QN75Q8DT%20Test&description=Short%20Product%20Description7&price=4000&img=https%3A%2F%2Fcrdms.images.consumerreports.org%2Ff_auto%2Cw_300%2Fprod%2Fproducts%2Fcr%2Fmodels%2F401740-60-inch-and-larger-tvs-samsung-qn75q8dt-10014881.png"`

JSON.parse is not appropriate for this.

Reply:
@PavelZaletski yeah! I understood. I just thought that you used plain text for your response.